homeassistant/components/onkyo/config_flow.py

```diff
@@ -0,0 +1,281 @@
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    Selector,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+)
+
+from .const import (
+    CONF_RECEIVER_MAX_VOLUME,
+    CONF_VOLUME_RESOLUTION,
+    CONF_VOLUME_RESOLUTION_DEFAULT,
+    DOMAIN,
+    OPTION_MAX_VOLUME,
+    OPTION_MAX_VOLUME_DEFAULT,
+    OPTION_SOURCE_PREFIX,
+    OPTION_SOURCES,
+    OPTION_SOURCES_ALLOWED,
+    OPTION_SOURCES_DEFAULT,
+)
+from .receiver import ReceiverInfo, async_discover, async_interview
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DEVICE = "device"
+
+STEP_CONFIGURE_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_VOLUME_RESOLUTION,
+            default=CONF_VOLUME_RESOLUTION_DEFAULT,
+        ): vol.In([50, 80, 100, 200]),
+        vol.Required(
+            OPTION_SOURCES, default=list(OPTION_SOURCES_DEFAULT.keys())
+        ): SelectSelector(
+            SelectSelectorConfig(
+                options=OPTION_SOURCES_ALLOWED,
+                multiple=True,
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+    }
+)
+
+
+class OnkyoConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Onkyo config flow."""
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._receiver_info: ReceiverInfo
+        self._discovered_infos: dict[str, ReceiverInfo] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        return self.async_show_menu(
+            step_id="user", menu_options=["manual", "eiscp_discovery"]
+        )
+
+    async def async_step_manual(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle manual device entry."""
+        errors = {}
+
+        if user_input is not None:
+            info: ReceiverInfo | None = None
+            host = user_input[CONF_HOST]
+            try:
+                info = await async_interview(host)
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                if info is None:
+                    errors["base"] = "cannot_connect"
+                else:
+                    self._receiver_info = info
+                    return await self.async_step_configure_receiver()
+
+        return self.async_show_form(
+            step_id="manual",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_HOST, default=""): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_eiscp_discovery(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Start eiscp discovery and handle user device selection."""
```

Comment:
Looking at this code, there seem to be some things missing/incorrect.

Was this tested? Does it actually work?

Reply:
Fixed myself. Can be marked resolved.